erasurecoding/src/lib.rs

```diff
@@ -0,0 +1,122 @@
+use libc::{c_int, size_t};
+use std::slice;
+
+/// Reed-Solomon encode function for FFI
+/// Takes sharded original data as byte array and produces recovery shards
+/// Parameters:
+/// - original_shards: number of original data shards
+/// - recovery_shards: number of recovery shards to generate
+/// - original_shards: input byte array containing flattened shards
+/// - shard_size: size of each shard in bytes
+/// - recovery_shards_out: buffer to store generated and flattened recovery shards
+/// Returns 0 on success, -1 on error
+#[no_mangle]
+pub unsafe extern "C" fn reed_solomon_encode(
+    original_shards_count: size_t,
+    recovery_shards_count: size_t,
+    shard_size: size_t,
```

Comment:
The general question - maybe we could just assign this to some new variable and not use conversion to `usize` during each usage inside the function? 😸 
Or can we just skip `as usize` each time? The decode function doesn't have it.

Reply:
Yeah I realized after we don't need it, forgot to update encode too. Good catch!